File: app/src/main - 副本/java/com/zmcursor/mydaliy/UI/Splash/SplashPresenter.java
package com.zmcursor.mydaliy.UI.Splash;

import com.zmcursor.mydaliy.Model.StoryDateHolder;
import com.zmcursor.mydaliy.MyAPP;

/**
 * Created by ZMcursor on 2018/6/2 0002.
 */

public class SplashPresenter {

    private SplashActivity splashActivity;
    private StoryDateHolder storyDateholder;

    public SplashPresenter(SplashActivity splashActivity) {
        this.splashActivity = splashActivity;
        storyDateholder = ((MyAPP) splashActivity.getApplication()).getStoryDateHolder();
    }

    public void initData() {
        storyDateholder.initData(((MyAPP) splashActivity.getApplication()).getHandler(), SplashActivity.msg_init);
    }

    public void saveData() {
        storyDateholder.saveData();
    }
}

File: app/src/main/java/com/zmcursor/mydaliy/UI/Splash/SplashPresenter.kt
package com.zmcursor.mydaliy.UI.Splash

import com.zmcursor.mydaliy.Model.StoryDateHolder
import com.zmcursor.mydaliy.MyAPP

/**
 * Created by ZMcursor on 2018/6/2 0002.
 */

class SplashPresenter(private val splashActivity: SplashActivity) {
    private val storyDateholder: StoryDateHolder?

    init {
        storyDateholder = (splashActivity.application as MyAPP).storyDateHolder
    }

    fun initData() {
        storyDateholder!!.initData((splashActivity.application as MyAPP).handler, SplashActivity.msg_init)
    }

    fun saveData() {
        storyDateholder!!.saveData()
    }
}
